conn.go

```diff
@@ -118,12 +118,12 @@ func (c *Conn) Streams() []*Stream {
 // Close closes this connection
 func (c *Conn) Close() error {
 	c.closeLock.Lock()
-	defer c.closeLock.Unlock()
-
 	if c.closed {
+		c.closeLock.Unlock()
```

Comment:
then those callers should say `go s.Close()` -- the semantics of `Close()` are that it _may_ be stuck waiting for it to fully close, and if something shouldnt be waiting, then it should really not be waiting